spot-contracts/contracts/RolloverVault.sol

```diff
@@ -497,24 +471,25 @@ contract RolloverVault is
             IERC20Burnable(address(perp)).burn(perpAmtToBurn);
         }
 
-        if (execRecover) {
-            // redeem perps for tranches and recover underlying
-            recover();
-        }
+        // Redeem perps for tranches and execute recover
+        _redeemPerps(perp);
+        recover();
```

Comment:
Wdyt about lazy recover?
```
if (underlyingAmtOut < underlying.balanceOf(this)) {
    recover();
}
underlying.transfer(...);
```

Case for the change: saves gas on the whole. Especially in cases where the vault is way overcollateralized as now, lots of these ops would never need recovers.

Case for keeping as-is: this function is mostly called by arbers, and they're already only calling if it's profitable, so just bundle in the positive externalities here. Result is slightly wider price bands, inline with gas costs for recover.

Reply:
As you noted offline, the perp redemption causes a lot of pollution to the tranche set. So makes sense for this caller to execute the cleanup. Let's keep as it is 👍